src/test/java/org/jenkinsci/plugins/docker/workflow/WithContainerStepTest.java

```diff
@@ -134,7 +147,68 @@
                 SemaphoreStep.success("wait/1", null);
                 WorkflowJob p = story.j.jenkins.getItemByFullName("prj", WorkflowJob.class);
                 WorkflowRun b = p.getLastBuild();
-                story.j.assertLogContains("Require method GET POST OPTIONS", story.j.assertBuildStatusSuccess(JenkinsRuleExt.waitForCompletion(b)));
+                story.j.assertLogContains("Require method GET POST OPTIONS", story.j.assertBuildStatusSuccess(story.j.waitForCompletion(b)));
+            }
+        });
+    }
+
+    @Issue("JENKINS-32943")
+    @Test public void fileCredentials() throws Exception {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                DockerTestUtil.assumeDocker();
+                File f = tmp.newFile("some-file");
+                FileUtils.write(f, "some-content");
+                FileItem fi = new FileParameterValue.FileItemImpl(f);
+                FileCredentialsImpl fc = new FileCredentialsImpl(CredentialsScope.GLOBAL, "secretfile", "", fi, fi.getName(), null);
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), fc);
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "prj");
+                p.setDefinition(new CpsFlowDefinition(
+                    "node {\n" +
+                    "  withDockerContainer('ubuntu') {\n" +
```

Comment:
`Docker.Image.inside` just calls `withDockerContainer`. This test suite is testing the step directly. `DockerDSLTest` covers the syntactic sugar.